### DISCUSSION TITLE
Automatically update volume text on keybind

### DISCUSSION THREAD

**Discussion by nathanielevan** [qtile/qtile#4324] [Q&A]
Hi, hopefully the discussion title is obvious enough. I've been trying to find a way so that when I increase or decrease volume (or mute), it immediately reflects on the volume widget without having to wait for a certain interval until it updates. I stumbled across force_update() but using it like `lazy.widget["pulsevolume"].force_update()` does not work. Any assistance would be much appreciated!

**Comment by ervinpopescu** [comment]
Check the widget name:

```sh
qtile cmd-obj -o bar top -f info |
python -c '''
import ast
import json
import sys

stdin = ast.literal_eval(sys.stdin.read())
print(json.dumps(stdin, indent=4))
''' |
jq -r --color-output '.widgets[].name' |
less
```

**Reply by nathanielevan** [reply]
I've got these:
```currentlayout
groupbox
prompt
windowname
pulsevolume
clock
systray
```
So it appears that pulsevolume already exists, there.
EDIT: Forgot to clarify that I'm running Arch Linux, qtile version 0.22.1-4 from the official repos

**Reply by ervinpopescu** [reply]
I'm really out of ideas, then. The logs say anything?

**Reply by nathanielevan** [reply]
Doesn't seem to say anything, sorry. I just looked around and found out that the force_update() seems to only work on ThreadPoolText widgets, which pulsevolume isn't? Am I understanding this correctly?

**Reply by ervinpopescu** [reply]
I believe you are right.

**Reply by elParaguayo** [reply]
The commands available to that widget are shown here: https://qtile.readthedocs.io/en/latest/manual/commands/api/widgets.html#pulsevolume

You could try `lazy.widget["pulsevolume"].function("self.update(self.poll())")`.
